libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1180,9 +1181,32 @@ void CompilerUtils::computeHashStatic()
 
 void CompilerUtils::storeStringData(bytesConstRef _data)
 {
-	//@todo provide both alternatives to the optimiser
+	// @todo provide both alternatives to the optimiser
 	// stack: mempos
-	if (_data.size() <= 128)
+	bool storeAsData = false;
+	if (_data.size() > 32)
+	{
+		unsigned numWords = (_data.size() + 31) / 32;
+		// @todo this does take the constant optimiser into account,
+		// which could reduce the costs of individual PUSHes
+		// NOTE: assumes the C++ compiler will optimise these statements
+		unsigned pushCost =
+			(eth::GasCosts::tier2Gas * numWords) + // PUSH value
```

Comment:
Please use the gas meter in conjunction with a sequence of assembly items.

Reply:
Looked into this now, it seems to be an overkill. `GasMeter` doesn't offer support for an item list and `PathGasMeter` seems to be heavy.

Did yoy mean using `PathGasMeter`?